virt-install

```diff
@@ -531,13 +531,19 @@ def set_explicit_guest_options(options, guest):
 
 def installer_detect_distro(guest, installer, osdata):
     try:
+        name = None
+
+        # OS name has to be set firstly whenever --os-variant is passed,
+        # otherwise it won't be respected when the installer creates the
+        # Distro Store.
+        if osdata.name and not osdata.is_auto:
+            name = osdata.name
+            guest.set_os_name(name)
+
```

Comment:
The changes in this file can be their own patch, they are a good to do independently I think

The name variable can be dropped.
The is_auto check here can be dropped too, `osdata.name` will be None if osdata.is_auto is True